src/control/server/faultdomain.go

```diff
@@ -92,5 +92,14 @@ func getFaultDomainFromCallback(callbackPath string) (*system.FaultDomain, error
 		return nil, FaultConfigFaultCallbackFailed(err)
 	}
 
-	return newFaultDomainFromConfig(string(output))
+	fd, err := newFaultDomainFromConfig(string(output))
+	if err != nil {
+		return nil, err
+	}
+
+	if fd.Empty() {
+		return nil, FaultConfigFaultCallbackEmpty
+	}
```

Comment:
Seems a little awkward to make the caller check for an empty domain rather than just returning the error from `newFaultDomainFromConfig()` if the input string is empty.

Reply:
I decided to check at this level since I thought a callback-specific error message would be helpful to the user. newFaultDomainFromConfig is shared by the fault path logic as well, although we shouldn't be hitting the empty case there, since the fault path is pulled directly from the yaml file. We don't bother with it if it's empty, and I don't think it's possible to end up with a whitespace string from the yaml parsing.

I was thinking of pre-checking, but figured the fault domain already has the logic to know if it's empty. I could just as well trim the output string and check it here though.

Reply:
OK, fair enough. I hadn't looked at the full context of the function. This is all internal, so I'm fine with the approach.

Reply:
Oops, didn't see your comment before pushing up a patch just now. I think it is a little clearer in the update anyway.